src/ess/isissans/mantidio.py

```diff
@@ -99,12 +100,29 @@ def from_data_workspace(
 
 
 def load_run(filename: Filename[RunType], period: Period) -> DataWorkspace[RunType]:
-    loaded = _mantid_simpleapi.Load(
-        Filename=str(filename), LoadMonitors=True, StoreInADS=False
-    )
+    # Loading many small files with Mantid is, for some reason, very slow when using
+    # the default number of threads in the Dask threaded scheduler (1 thread worked
+    # best, 2 is a bit slower but still fast). We can either limit that thread count,
+    # or add a lock here, which is more specific.
+    with load_run.lock:
+        loaded = _mantid_simpleapi.Load(Filename=str(filename), StoreInADS=False)
+    try:
+        loaded = _mantid_simpleapi.Load(
+            Filename=str(filename), LoadMonitors=True, StoreInADS=False
+        )
+    except TypeError:
+        # Not loaded using LoadEventNexus, so LoadMonitor option is not available.
+        loaded = _mantid_simpleapi.Load(Filename=str(filename), StoreInADS=False)
```

Comment:
Why do you call Load again after the lock?

Reply:
Oops, should have been inside the lock. See update,